examples/dx-bwa-mem/src/dx-bwa-mem.sh

```diff
@@ -0,0 +1,87 @@
+#!/bin/bash
+# dx-bwa-mem 0.0.1
+# Generated by dx-app-wizard.
+#
+# Basic execution pattern: Your app will run on a single machine from
+# beginning to end.
+#
+# Your job's input variables (if any) will be loaded as environment
+# variables before this script runs.  Any array inputs will be loaded
+# as bash arrays.
+#
+# Any code outside of main() (or any entry point you may add) is
+# ALWAYS executed, followed by running the entry point itself.
+#
+# See https://wiki.dnanexus.com/Developer-Portal for tutorials on how
+# to modify this file.
+
+main() {
+
+    echo "Value of genome: '$genome'"
+    echo "Value of read_one_fastq: '$read_one_fastq'"
+    echo "Value of read_two_fastq: '$read_two_fastq'"
+
+    # The following line(s) use the dx command-line tool to download your file
+    # inputs to the local file system using variable names for the filenames. To
+    # recover the original filenames, you can use the output of "dx describe
+    # "$variable" --name".
+
+    dx download "$read_one_fastq" 
+
+    dx download "$read_two_fastq" 
+
+    # Get reference files for BWA-MEM
+    dx download -r project-F5444K89PZxXjBqVJ3Pp79B4:/global/reference/Homo_sapiens/${genome}/BWA
+
+    # Install conda
+    pypath=$PYTHONPATH
+    orig_path=$PATH
+    export PYTHONPATH=
+    wget "https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh" -O miniconda.sh 
+    chmod a+x miniconda.sh 
+       ./miniconda.sh -b -p /opt/conda/ 
+       rm miniconda.sh
+
+    PATH=/opt/conda/bin:$PATH
+
+    conda update -n base -c defaults conda -y && \
+       conda install \
+       -c conda-forge \
+       -c bioconda \
+       bwa==0.7.17 \
+       samtools==1.10 \
+       -y && \
+       conda clean --all -y
+
+    # Fill in your application code here.
+    #
+    # To report any recognized errors in the correct format in
+    # $HOME/job_error.json and exit this script, you can use the
+    # dx-jobutil-report-error utility as follows:
+    #
+    #   dx-jobutil-report-error "My error message"
+    #
+    # Note however that this entire bash script is executed with -e
+    # when running in the cloud, so any line which returns a nonzero
+    # exit code will prematurely exit the script; if no error was
+    # reported in the job_error.json file, then the failure reason
+    # will be AppInternalError with a generic error message.
+    bwa mem -t 4 BWA/${genome}.fa $read_one_fastq_name $read_two_fastq_name | samtools view -Sb - > out.bam
+
+    # The following line(s) use the dx command-line tool to upload your file
+    # outputs after you have created them on the local file system.  It assumes
+    # that you have used the output field name for the filename for each output,
+    # but you can change that behavior to suit your needs.  Run "dx upload -h"
+    # to see more options to set metadata.
+    export PYTHONPATH=$pypath
+    export PATH=$orig_path
+
+    output_bam=$(dx upload *.bam --brief)
```

Comment:
```suggestion
    output_bam=$(dx upload out.bam --brief)
```
Not sure if it will error out if there are multiple bam files outputted, but may confuse others that use this as a template.